### DISCUSSION TITLE
How to create an account via smart contract?

### DISCUSSION THREAD

**Discussion by romeobuilderotti** [farcasterxyz/protocol#159] [Q&A] [answered]
What tx should I send to register?

**Comment by horsefacts** [comment, marked as answer]
You can call [`IdGateway.register`](https://docs.farcaster.xyz/reference/contracts/reference/id-gateway#register) to create an ID and [`KeyGateway.add`](https://docs.farcaster.xyz/reference/contracts/reference/key-gateway#add) to add a signer public key.

Depending on how you intend to interact with the protocol, you may want to register this account to a fresh address. If you want to use Warpcast, you'll need to import the seed phrase of the account that owns your FID, so it's a good idea to use a single purpose address rather than a shared wallet you use for other types of transactions.

**Reply by sds** [reply]
You can also check out this tutorial: https://docs.farcaster.xyz/developers/guides/accounts/create-account
